src/main/java/com/zaxxer/hikari/HikariConfig.java

```diff
@@ -880,11 +882,17 @@ protected void loadProperties(String propertyFileName)
 
    private int generatePoolNumber()
    {
-      // Pool number is global to the VM to avoid overlapping pool numbers in classloader scoped environments
-      synchronized (System.getProperties()) {
-         final int next = Integer.getInteger("com.zaxxer.hikari.pool_number", 0) + 1;
-         System.setProperty("com.zaxxer.hikari.pool_number", String.valueOf(next));
-         return next;
+      try {
```

Comment:
Instead of catching AccessControlException why not something like this..?

```java
 Integer next = AccessController.doPrivileged(new PrivilegedAction<Integer>() {
                @Override
                public Integer run() {
                    // Pool number is global to the VM to avoid overlapping pool numbers in classloader scoped environments
         synchronized (System.getProperties()) {
            final int next = Integer.getInteger("com.zaxxer.hikari.pool_number", 0) + 1;
            System.setProperty("com.zaxxer.hikari.pool_number", String.valueOf(next));
            return next;
         }
                }
            });
```
If null fallback to static atomic integer.